Angular/Angular guidelines and best practices/Formatting, naming and best practices.md

```diff
@@ -675,6 +675,66 @@ Template:
 ></my-app-article-details>
 ```
 
+## Avoid concatenation of translation keys
+
+Usual requirement for applications could be that the application needs to be localized. Regardless of the library which is used for the fetching of localized values,
+we should avoid concatenation of keys used for fetching translations. One of the issues is keeping the file with translations clean, it could prove very tiresome and hard even, to keep that file in sync with translations that are used throughout the application when you have to search for partial strings instead of searching for the whole string. One of the most tempting cases for concatenation is when you iterate over some enum and match those values with the ones received from the backend service. 
```

Comment:
```suggestion
we should avoid concatenation of translation keys. Concatenating translation keys makes managing of translation keys harder. It could prove very tiresome to keep the translations file in sync with translations that are used throughout the application, if you have to search for partial strings instead of searching for the whole string. One of the most tempting cases for using concatenation is when you iterate over some enum and match those values with the ones received from the backend service.
```